src/librustc_codegen_llvm/common.rs

```diff
@@ -86,6 +86,8 @@ impl Funclet<'ll> {
 
 impl BackendTypes for CodegenCx<'ll, 'tcx> {
     type Value = &'ll Value;
+    type FuncId = &'ll Value;
```

Comment:
Maybe we should have a newtype of `Value` to force this to be correctly typed (LLVM has a class for functions which IIRC is a subclass of `GlobalValue`). But this is a low-priority cleanup.